packages/signature-verifier-api/src/client.rs

```diff
@@ -0,0 +1,44 @@
+use cosmwasm_schema::serde::de::DeserializeOwned;
+use cosmwasm_std::{to_binary, Addr, HexBinary, QuerierWrapper, QueryRequest, Uint64, WasmQuery};
+use error_stack::{Result, ResultExt};
+
+use crate::msg::QueryMsg;
+
+pub struct SignatureVerifier<'a> {
+    pub address: Addr,
+    pub querier: QuerierWrapper<'a>,
+}
+
+impl SignatureVerifier<'_> {
+    fn query<U: DeserializeOwned + 'static>(&self, msg: &QueryMsg) -> Result<U, Error> {
+        self.querier
+            .query(&QueryRequest::Wasm(WasmQuery::Smart {
+                contract_addr: self.address.to_string(),
+                msg: to_binary(msg).expect("msg should always be serializable"),
+            }))
+            .change_context(Error::QuerySignatureVerifier)
+    }
+
+    pub fn verify_signature(
+        &self,
+        signature: HexBinary,
+        message: HexBinary,
+        public_key: HexBinary,
+        signer_address: String,
+        session_id: Uint64,
+    ) -> Result<bool, Error> {
+        self.query(&QueryMsg::VerifySignature {
```

Comment:
What happens when the query message in JSON has more fields than the quering contract expects? Does it force deserialize all bytes and therefore would fail the query? Asking cuz we need to understand what kind of impact it would be to add new fieds to this query msg when some callbacks already exist.

Reply:
good point, the deserialization will fail. I'll think a bit more about this